src/utils/plotSpecs/generateMitochondrialContentSpec.js

```diff
@@ -139,4 +139,56 @@ const generateSpec = (config, plotData) => {
   };
 };
 
-export default generateSpec;
+const filterCells = (cellSets, selectedCellSet) => {
+  let newCellSets = cellSets.hierarchy.find(
+    (rootNode) => rootNode.key === selectedCellSet,
+  )?.children || [];
+
+  // Build up the data source based on the properties. Note that the child nodes
+  // in the hierarchy are /objects/ with a `key` property, hence the destructuring
+  // in the function.
+  newCellSets = newCellSets.flatMap(({ key }) => {
+    const cells = Array.from(cellSets.properties[key].cellIds);
+
+    return cells.map((cellId) => ({
+      cellId,
+      cluster: cellSets.properties[key].name,
+      color: cellSets.properties[key].color,
+    }));
+  });
+
+  return newCellSets;
+};
+
+const generateData = (
+  cellSets,
+  selectedSample,
+  plotData,
+  embeddingData,
+) => {
+  // the result of the map on embedding data contains non-consecutive indices (aka empties)
+  // filter removes missing elements in the array thus effectively reindexing the resulting array
+  // into consecutive indexes which is the format expected by Vega.
+  // If we have a specific sample then filter will act normally, in case of using all samples
+  // we just provide a dummy filter function.
```

Comment:
The comment does not seem to match the code.  Does the filter need to different from the one in `generateEmbeddingContinuousSpec.js`? Can they maybe be refactored into a single function that can be used for different plots?

Reply:
Mitchondrial content plot and specs aren't used anymore. The same with doublet scores. The plot elements have been removed. I'll remove these files also